fix pep8/flake8 errors and warnings

sbellem opened the pull request:
Closes amiller/honeybadgerbft#33

sbellem committed:
fix: pep8 E305

expected 2 blank lines after end of function or class

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E703

statement ends with a semicolon

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E211

whitespace before ‘(‘

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E302

expected 2 blank lines, found 0

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes<Paste>

sbellem committed:
fix: pep8 E201, E202

E201: whitespace after ‘(‘
E202: whitespace before ‘)’

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E231

missing whitespace after ‘,’, ‘;’, or ‘:’

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E262

inline comment should start with ‘# ‘

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E203

whitespace before ‘:’

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E265

E265: block comment should start with ‘# ‘

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E261

at least two spaces before inline comment

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E272

multiple spaces before keyword

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E221

multiple spaces before operator

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E225

missing whitespace around operator

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E266

too many leading ‘#’ for block comment

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E227

missing whitespace around bitwise or shift operator

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E303

too many blank lines (2 or 3)

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes<Paste>

sbellem committed:
fix: pep8 E701

multiple statements on one line (colon)

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E712

comparison to True should be ‘if cond is True:’ or ‘if cond:’

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 E731

do not assign a lambda expression, use a def

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes<Paste>

sbellem committed:
Ignore pep8 error E741 for now

sbellem committed:
fix: pep8 E114

indentation is not a multiple of four (comment)

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 W291

trailing whitespace

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 W391

blank line at end of file

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pep8 W604

backticks are deprecated, use ‘repr()’

Ref: https://pep8.readthedocs.io/en/latest/intro.html#error-codes

sbellem committed:
fix: pyflakes F401

module imported but unused

Ref: http://flake8.pycqa.org/en/latest/user/error-codes.html#error-violation-codes

sbellem committed:
fix: pyflakes F403

‘from module import *’ used; unable to detect undefined names

Ref: http://flake8.pycqa.org/en/latest/user/error-codes.html#error-violation-codes

sbellem committed:
fix: pyflakes F841

local variable name is assigned to but never used

Ref: http://flake8.pycqa.org/en/latest/user/error-codes.html#error-violation-codes

sbellem committed:
Set max_line_length flake8 config to 99

sbellem committed:
Mark test for not yet fixed issue #59 with xfail

sbellem committed:
Add flake8 check to travis

sbellem committed:
Allow codecov project/patch to drop by 2%/1%

This is added because codecov is currently reporting a small
coverage decrease despite the fact that computing the coverage
locally does not show any such decrease. So for the time being
we can tolerate small decreases especially given that no such
decrease is observed locally.

amiller reviewed (approved):
Looks good, viewed the changes to confirm they match style and don't functionally change.

I think it is worth an exception when clarifying visual parallel structure, particularly when off by one errors are prone. Examples:
```
         num = reduce(mul, [0 - jj - 1 for jj in S if jj != j], ONE)
-        den = reduce(mul, [j - jj     for jj in S if jj != j], ONE)
```

```
-    K               = N - 2 * f  # Need this many to reconstruct
-    EchoThreshold   = N - f      # Wait for this many ECHO to send READY
-    ReadyThreshold  = f + 1      # Wait for this many READY to amplify READY
```

sbellem committed:
Tolerate some pep8 errors for matrix-like code

E221 and E272 errors are tolerated so that the code can be aligned in a
matrix like structure which can in turn help detect small errors that
would otherwise be easy to miss.

See comment by @amiller: https://github.com/initc3/HoneyBadgerBFT-Python/pull/1#pullrequestreview-122765196

sbellem commented:
> I think it is worth an exception when clarifying visual parallel structure, particularly when off by one errors are prone.

Oh ok sure. We can simply add `# noqa: E2xx` at the end a line to ignore the pep8 error.

See new commit 8988350
